### DISCUSSION TITLE
Improvements proposal ideas

### DISCUSSION THREAD

**Discussion by AbdelStark** [keep-starknet-strange/unruggable.meme#221] [Ideas]
Give your improvement proposal ideas.

**Comment by starknetcableguy** [comment]
In it's current form, Unruggable is a great tool for quickly and easily deploying memes 🤌

I think these two things could take it to the next level:

1) Governance

Implimenting a 'governance' feature could help extend the lifecycle of memes & create a demand-sink for the token.

'veUnruggable' could be something like:
- Deployer of [$UNRUG] has the option to allocate X% of supply to a 'Community Wallet'
- Holders can lock [$UNRUG] for a set amount of time to receive  [veUNRUG]
- [veUNRUG] is non-transferable, and longer you lock, the more you receive
- Community Wallet is controlled via [veUNRUG] voting

This could be small things such as selecting memecoin contest winners, or bigger things such as awarding grants.

2) Liquidity

We've all seen the Starknet memecoin charts — liquidity is a problem.

IMO an Unruggable DEX with the following features would help solve the liquidity problem:

- Anyone can allocate a certain amount of an unruggable memecoin to emit to a specific pool (min allocation >1% of total supply or something like this)
- Not only would this reward long-term holders for providing liquidity, but also memecoins could form alliances and decide to emit both their tokens to their joint liquidity pool (8/PAL for example).
- The trading fees for this DEX could be used to buy $STRK
- DEX could emit $STRK to memecoin LP's based on the total trading volume of the memecoin on the DEX (with a cap at something like 20% of total emissions per pool).

**Comment by 0xQwerty0** [comment]
# Making Unruggable Even Better: My proposal

Hello,

I've been having a blast launching my memecoins with your platform, and I've got a few ideas that could make things even cooler for everyone. Here’s what I'm thinking:

## Cool Stuff We Can Do Without Any Fancy Backend Stuff

### Direct Buy Page

- How about a page where you can just zip in and grab some coins, kind of like what [This zapper](https://zapper.sakupen.org/) does? It could make getting into a coin super easy and fast.
- It’d make it simpler for folks to jump in and buy coins, which means more action for all the coins on the platform.

### Drop an Unruggable SDK

- Imagine if we had a toolkit with all the building blocks from Unruggable ready to go. Developers could grab it and start building their own stuff right on top.
- It'd cut down on having to reinvent the wheel and could spark all kinds of new projects. Plus, it makes our base stronger when everyone builds on a common foundation.

## Features That Need Some Backend

### Leaderboard for Top Performing Memes

- How cool would it be to see which memecoins are killing it based on ROI from their launch? A little competition never hurts!
- Gives everyone a clear view of what’s hot, helping users make smarter picks and seeing which strategies work best.

### Trendy Memes on the Landing Page

- Let's jazz up the landing page by showcasing the hottest memes right now, based on how much buzz they’re getting.
- It keeps the site fresh and engaging. Plus, it’s great for newcomers to see what’s trending at a glance and get in on the action.
